### PR TITLE
feat(ColorPicker): close panel when click clear button

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

close panel when click clear button

- https://element.eleme.cn/#/zh-CN/component/color-picker

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | close panel when click clear button |
| 🇨🇳 Chinese |  点击清除按钮时隐藏面板  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 8a8b820</samp>

Refactored `ColorPicker` component and updated its test case. Added a `clearColor` feature that allows users to reset the color and hide the panel.

### <samp>🤖 Generated by Copilot at 8a8b820</samp>

*  Add `clearColor` feature to `ColorPicker` component to allow user to clear color selection ([link](https://github.com/ant-design/ant-design/pull/42406/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fL9-R10), [link](https://github.com/ant-design/ant-design/pull/42406/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fR138-R143))
* Update test case for `clearColor` feature in `index.test.tsx` file ([link](https://github.com/ant-design/ant-design/pull/42406/files?diff=unified&w=0#diff-aa06460351e8630f305cf6509e8137984e0f167ab2560da22874033189234fb4L86-R90))

### CODE REVIEW

File: components/color-picker/ColorPicker.tsx [hunk 1]
```diff
@@ -135,6 +135,12 @@ const ColorPicker: CompoundedComponent = (props) => {
     onFormatChange,
   };
 
+  useEffect(() => {
+    if (clearColor) {
```
Comment (afc163) [inline components/color-picker/ColorPicker.tsx:139]: clearColor 是个动词，表示什么状态？

Comment (kiner-tang) [reply components/color-picker/ColorPicker.tsx:139]: 确实，我顺带改一下变量名吧

File: components/color-picker/ColorPicker.tsx [hunk 2]
```diff
@@ -128,13 +128,19 @@ const ColorPicker: CompoundedComponent = (props) => {
...
     presets,
     format,
     onFormatChange,
   };
 
+  useEffect(() => {
+    if (isClearColor) {
+      setPopupOpen(false);
+    }
+  }, [isClearColor]);
```
Comment (afc163) [inline components/color-picker/ColorPicker.tsx:142]: 不如叫 `colorCleared`

Comment (kiner-tang) [reply components/color-picker/ColorPicker.tsx]: 好

**Review by afc163 — APPROVED** [review]

**Pull request merged**